src/libpmemobj/palloc.c

```diff
@@ -891,6 +915,9 @@ palloc_defrag(struct palloc_heap *heap, uint64_t **objv, size_t objcnt,
 			goto err_objvp;
 	}
 
+	if (current_object_sequence > longest_object_sequence)
```

Comment:
you can just remove the if inside the for loop now.

Reply:
If I would do that, then we would use only the last sequence.